### PR TITLE
Fixing random_step function

### PR DESCRIPTION
**Pull request by ClaudiusGruner**
In the calculation of the noise scaling, there was a mixup of division and multiplication. According to Algorithm 1 in the paper, it should be multiply Euler step and variance instead of division.

### CODE REVIEW

**Comment by gaspardol** [comment]
Thank you for having a look at the code.

The proposed change misunderstands the implementation of the noise scaling. In the current implementation, the gradient is later multiplied by the learning rate during optimizer.step(), as is standard for SGD. The original code correctly applies the scaling sqrt(lr), since multiplying the noise by sqrt(1/lr) before optimizer.step() results in an effective scaling of lr * sqrt(1/lr) = sqrt(lr). Thus, no change is needed and the apparent division is intentional and correct.
